src/transport/msg_providers/data_bus.py

```diff
@@ -0,0 +1,281 @@
+import abc
+import logging
+from collections import deque
+from enum import Enum
+from typing import List, Optional
+
+import variables
+from eth_account.account import VRS
+from eth_typing import HexStr
+from schema import Schema
+from transport.msg_providers.common import BaseMessageProvider
+from transport.msg_providers.rabbit import MessageType
+from transport.msg_types.deposit import DepositMessage
+from transport.msg_types.pause import PauseMessage
+from transport.msg_types.ping import PingMessageDataBus
+from transport.msg_types.unvet import UnvetMessage
+from utils.bytes import bytes_to_hex_string
+from web3 import Web3
+from web3._utils.events import get_event_data
+from web3.exceptions import BlockNotFound
+from web3.types import FilterParams, LogReceipt
+
+logger = logging.getLogger(__name__)
+
+_MESSAGE = {
+    'anonymous': True,
+    'inputs': [
+        {'indexed': True, 'name': 'eventId', 'type': 'bytes32'},
+        {'indexed': True, 'name': 'sender', 'type': 'address'},
+        {'indexed': False, 'name': 'data', 'type': 'bytes'},
+    ],
+    'name': 'Message',
+    'type': 'event',
+}
+
+UNVET_V1_DATA_SCHEMA = '(uint256,uint256,bytes32,uint256,bytes,bytes32,bytes32,(bytes32))'
+PING_V1_DATA_SCHEMA = '(uint256,(bytes32))'
+DEPOSIT_V1_DATA_SCHEMA = '(bytes32,uint256,uint256,bytes32,bytes,uint256,(bytes32))'
+PAUSE_V2_DATA_SCHEMA = '(bytes32,uint256,uint256,bytes32,bytes,uint256,(bytes32))'
+PAUSE_V3_DATA_SCHEMA = '(uint256,bytes,(bytes32))'
+
+
+def signature_to_r_vs(signature: bytes) -> tuple[VRS, VRS]:
+    # 0 byte - 0x
```

Comment:
0x is just indication for hexstr, if we read bytes from contract there is no reason to encode `0x` to signature

Reply:
Done